Mathlib/Algebra/MvPolynomial/Degrees.lean

```diff
@@ -524,6 +524,53 @@ theorem totalDegree_rename_le (f : σ → τ) (p : MvPolynomial σ R) :
 
 end TotalDegree
 
+section degreesLE
+variable {s t : Multiset σ}
+
+variable (R σ s) in
+/-- The submodule of multivariate polynomials of degrees bounded by a monomial `s`. -/
+def degreesLE : Submodule R (MvPolynomial σ R) where
+  carrier := {p | p.degrees ≤ s}
+  add_mem' {a b} ha hb := by classical exact degrees_add_le.trans (sup_le ha hb)
+  zero_mem' := by simp
+  smul_mem' c {x} hx := by
+    dsimp
+    rw [Algebra.smul_def]
+    refine degrees_mul_le.trans ?_
+    simpa [degrees_C] using hx
+
+@[simp] lemma mem_degreesLE : p ∈ degreesLE R σ s ↔ p.degrees ≤ s := Iff.rfl
+
+variable (s t) in
+lemma degreesLE_add : degreesLE R σ (s + t) = degreesLE R σ s * degreesLE R σ t := by
+  classical
+  rw [le_antisymm_iff, Submodule.mul_le]
+  refine ⟨fun x hx ↦ ?_, fun x hx y hy ↦ degrees_mul_le.trans (add_le_add hx hy)⟩
+  rw [x.as_sum]
+  refine sum_mem fun i hi ↦ ?_
+  replace hi : i.toMultiset ≤ s + t := (Finset.le_sup hi).trans hx
+  let a := (i.toMultiset - t).toFinsupp
+  let b := (i.toMultiset ⊓ t).toFinsupp
+  have : a + b = i := Multiset.toFinsupp.symm.injective (by simp [a, b, Multiset.sub_add_inter])
+  have ha : a.toMultiset ≤ s := by simpa [a, add_comm (a := t)] using hi
+  have hb : b.toMultiset ≤ t := by simp [b, Multiset.inter_le_right]
```

Comment:
Seems like [Multiset.sub_add_inter](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Multiset/Basic.html#Multiset.sub_add_inter) and [Multiset.sub_le_iff_le_add](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Multiset/Basic.html#Multiset.sub_le_iff_le_add) are all you need and Finsupp is not needed.

Reply:
What do you mean? I want `a` and `b` to be finsupps because that's what `degrees_monomial` expects

Reply:
Oh, I thought you just want to show if `u ≤ s + t` then there exists `us` and `ut` such that `us ≤ s`, `ut ≤ t` and `u = us + ut`. Looks like something else is involved. I'll investigate later.